tests/framework/widgets/ListViewTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace yiiunit\framework\widgets;
+
+use yii\data\ArrayDataProvider;
+use yii\widgets\ListView;
+
+/**
+ * @group widgets
+ */
+class ListViewTest extends \yiiunit\TestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->mockApplication();
+    }
+
+    public function testEmptyListShown()
+    {
+        $dataProvider = new ArrayDataProvider([
+            'allModels' => [],
+        ]);
+
+        ob_start();
+        echo ListView::widget([
+            'dataProvider' => $dataProvider,
+            'showOnEmpty' => false,
+            'emptyText' => "Nothing at all",
+        ]);
+        $actualHtml = ob_get_clean();
+
+        $this->assertTrue(strpos($actualHtml, "Nothing at all") !== false, "displays the empty message");
```

Comment:
these assertions are a but difficult to read. Can you use `assertContains` or `assertNotContains`